File: Divide and Conquer/quicksort.py
# Implementation of quick sort in python

def partition(l,left,right):
       pivot=l[right]
       i=-1
       for j in range(0,right):
              if l[j]<=pivot:
                     (l[i+1],l[j])=(l[j],l[i+1])
                     i=i+1
       (l[i+1],l[right])=(l[right],l[i+1])
       return i+1

def quicksort(l,left,right):
       print("Hello")
       if right>left:
              q=partition(l,left,right)
              quicksort(l,left,q-1)
              quicksort(l,q+1,right)

l=[]
l=list(map(int,input().split()))
n=len(l)
print(l)
r=0
quicksort(l,r,n-1)
print(l)

File: Divide and Conquer/Quick sort.cpp
#include<iostream>
using namespace std;

// This is the code for quick sort in c++
void swap(int *p,int *q)
{
	int temp;
	temp=*p;
	*p=*q;
	*q=temp;
}
int partition(int *arr,int left,int right)
{
	int pivot=arr[right];
	int i=left-1;
	for (int j=left;j<right;j++)
	{
		if (arr[j]<=pivot)
		{
			i++;
			swap(&arr[j],&arr[i]);
		}
	}
	swap(arr[right],arr[i+1]);
	return (i+1);
}
void quicksort(int *arr, int left,int right)
{
	if( right>left)
	{
		int q=partition(arr,left,right);
		quicksort(arr,left,q-1);
		quicksort(arr,q+1,right);
	}
}
int main()
{
	int n;
	cout<<"Enter the number of elements "<<endl;
	cin>>n;
	int arr[n];
	cout<<"Enter elements"<<endl;
	for(int i=0;i<n;i++)
	{
		cin>>arr[i];
	}
	quicksort(arr,0,n-1);
	for(int i=0;i<n;i++)
	{
		cout<<arr[i]<<" ";
	}
}
